src/agenda.ts

```diff
@@ -22,56 +25,64 @@ export class Person implements Searchable {
             return false
         }
     }
-
-    if(vesal mår inte bra) {
-        try {
-            hål i huvudet på vesal
-        }
-        catch() {
-            mår bra
-        }
-    }
-
 }
 
-export class Pet implements Searchable{
-    petName: string;
-    petType: string;
-    petOwner: string;
+class Pet implements Searchable{
 
-    constructor (name: string, type: string, owner: string){
-        this.petName=name;
-        this.petType=type;
-        this.petOwner=owner;
-    }
+    constructor (private petName: string, private petType: string, private petOwner: string){}
 
     search(searchValue: string) {
         if(searchValue === this.petType){
-            return true
+            return true;
         } else {
-            return false
+            return false;
         }
     }
-
 }
 
 
-export class Agenda {
+declare function require(name:string);
+let fs = require("fs");
+
+export class Agenda implements Serializable, Creatable{
+
     public contacts: Contact[] = [];
-    
+
+    public create(objectType, param1, param2, param3){
+        let newContact: any;
+        if(objectType === 'pet'){
+            let newContact = new Pet(param1, param2, param3);
+            return newContact;
+        }else if(objectType === 'person'){
+            let newContact = new Person(param1, param2, param3);
+            return newContact;
+        }else{
+            console.log('Contact type is missing');
+        }
+        return newContact;
+    }
+
     public addContact(contact: Contact){
         this.contacts.push(contact);
         return this.contacts;
     }
 
     public searchInAgenda(searchValue: string){
-
         function search(contact: Searchable){
             return contact.search(searchValue);
         }
-
         return this.contacts.filter(search);
+    }
 
+    public serialize(){
+        let saveObj = JSON.stringify(this.contacts);
+        fs.writeFileSync('./contacts.json', saveObj);
+        return saveObj;
     }
 
-}
+    public deserialize(folderPath){
```

Comment:
Not really sure what you are doing here. It only print the contents of the file.
Each class (Person, Pet) should have `serializable` and `deserializable`. When you obtain the text from the file, you need to figure out (in some way) which class to use (Person or Pet) and use their own deserializable. Then attach it to the Agenda.